### ISSUE TITLE
Add support for prefers color scheme

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyone#410]
As our goal is to create an accessible theme, the theme should support prefers-color-scheme.

https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme

I suggest adding a dark scheme at least, but it does not need to be black, using one of the existing darker color palettes would be enough.

**Comment by aristath** [comment]
I've wanted this for a looong time... Slightly related discussion on Gutenberg: https://github.com/WordPress/gutenberg/issues/24368

The problem that I haven't figured out yet is whether this should be opt-in or opt-out, and how much we want to automate it. For example if the user has selected to allow auto-dark-scheme and they have set a color like `#f2f2f2` for their background, then how do we calculate the dark-mode background color?
The best choice I've found so far is to convert the hex color they selected to HSL, and reverse the lightness (255 - L) to get the corresponding dark color.
It works and I did it a while back on an experimental theme I was working on, so we can definitely do it both in PHP for the frontend and in JS for the customizer.
That then leaves blocks where the user has selected an override background-color for that block. Should the same process be repeated on a per-block basis? :thinking:

**Comment by carolinan** [comment]
My thought was to just to preselect a darker background color no matter what the site owner has set, and the text color would be adjusted automatically.

And that blocks with specific colors set in the content, should keep those set colors.

The current defaults could be used as the light version.

**Comment by melchoyce** [comment]
Can we use the color palette output by the `#28303d` background color option as a base for dark mode support?

![image](https://user-images.githubusercontent.com/2846578/96202707-e7b0ca00-0f2d-11eb-88fc-558fdfc33744.png)
